editor/filesystem_dock.h

```diff
@@ -369,6 +369,9 @@ class FileSystemDock : public VBoxContainer {
 	static void _bind_methods();
 
 public:
+	const HashMap<String, Color> &get_folder_colors() const;
+	Dictionary get_assigned_folder_colors() const;
```

Comment:
Shouldn't I change this to be a const reference as well?
```suggestion
	const Dictionary &get_assigned_folder_colors() const;
```

Reply:
No dictionary is shared

Reply:
> No dictionary is shared

Uh? What do you mean? Sorry, I'm not very familiar with the codebase yet.

Reply:
Copying a dictionary just makes a reference, it doesn't copy the data, if you make a copy and change it the original changes too

Reply:
It could still be const reference, to avoid increasing reference counter. Though the gain is probably minimal.